ModifiedBayes.py

```diff
@@ -0,0 +1,168 @@
+		dataFrame = dataSet.dataFrame
+		print("c+s- count:", CHigherSLowerCount)
+		print("c-s- count:", CLowerSLowerCount)
+		print("c+s+ count:", CHigherSHigherCount)
+		print("c-s+ count:", CLowerSHigherCount)
+		print("bayes classification column c+s- count: ", self.countIntersection(dataFrame, dataSet.protectedAttribute, higherOrLowerSensitiveAttributeDict["lower"], "Bayes Classification", higherOrLowerClassificationDict["higher"]))
+		print("bayes classification column c-s- count: ", self.countIntersection(dataFrame, dataSet.protectedAttribute, higherOrLowerSensitiveAttributeDict["lower"], "Bayes Classification" , higherOrLowerClassificationDict["lower"]))
+		print("bayes classification column c+s+ count: ", self.countIntersection(dataFrame, dataSet.protectedAttribute, higherOrLowerSensitiveAttributeDict["higher"], "Bayes Classification", higherOrLowerClassificationDict["higher"]))
+		print("bayes classification column c-s+ count: ", self.countIntersection(dataFrame, dataSet.protectedAttribute, higherOrLowerSensitiveAttributeDict["higher"], "Bayes Classification", higherOrLowerClassificationDict["lower"]))
+
+	'''Space saving function for modify() that prints out probabilities'''
+	def printProbabilities(self, CHigherSLower, CLowerSLower, CHigherSHigher, CLowerSHigher):
+		print("c+s-:", CHigherSLower)
+		print("c-s-:", CLowerSLower)
+		print("c+s+:", CHigherSHigher)
+		print("c-s+:", CLowerSHigher)
+
+		
+	def train(self, dataSet, CHigher):
+		ModifiedNaive.train(self, dataSet, self.model)
+		self.modify(dataSet, CHigher)
+		
+		
+		
+	'''Trains and classifies the dataset '''
+	def modify(self, dataSet, CHigher):
+		dataFrame = dataSet.dataFrame
+		protected = dataSet.protectedAttribute
+		groundTruth = dataSet.trueLabels
+		sensitiveAttributeModelIndex = dataSet.headers.index(protected) #need to know index of sensitive attribute in the model
+
+		dataFrame = self.classify(dataSet)
+
+		#Assign dictionary values based on CHigher parameter
+		classesList = self.getAttributeCategories(dataFrame, dataSet.trueLabels)
+		higherOrLowerClassificationDict = {}
+		self.assignClassifications(higherOrLowerClassificationDict, CHigher, classesList)
+
+		#Assign the two sensitive attribute categories as S+ and S-
+		higherOrLowerSensitiveAttributeDict = {}
+		self.assignSensitivity(dataSet, dataFrame, higherOrLowerSensitiveAttributeDict)
+
+		#calculate the number of people in the dataset that are actually classified as C+ (in the ground truth column - the real number from the data)
+		actualNumPos = self.calculateNumPos(dataFrame, groundTruth, higherOrLowerClassificationDict)
+		print("The actualNumPos is: ", actualNumPos)
+
+		#Compute counts for C+S-,C-S+,C+S+,and C-S- based on counts from the original groundTruth column
+		CHigherSLowerCount = self.countIntersection(dataFrame, protected, higherOrLowerSensitiveAttributeDict["lower"], groundTruth, higherOrLowerClassificationDict["higher"])
```

Comment:
every time I read the variable names I cackle a little, but also they're descriptive so its good haha